### PR TITLE
Task 2 / Implemented Registration flow

### PR DESCRIPTION
**Pull request by Asmita04**

### CODE REVIEW

File: src/main/java/com/bluesky/bankapp/BankApplication.java [hunk 1]
```diff
@@ -10,58 +12,179 @@
  * */
 public class BankApplication {
 
+    public Map<String, User> map;
```
Comment (harry8906) [inline src/main/java/com/bluesky/bankapp/BankApplication.java:43]: User proper name, should be userMap

File: src/main/java/com/bluesky/bankapp/BankApplication.java [hunk 2]
```diff
@@ -10,58 +12,179 @@
...
+    public void addUserAccount(@NotNull User user) {
+        String adhaarNo = user.getAdhaarNo();
+        if (!accountNumbers.containsKey(adhaarNo)) {
+            accountNumbers.put(adhaarNo, 1);
+        } else {
+            int sequenceNumber = accountNumbers.get(adhaarNo) + 1;
+            accountNumbers.put(adhaarNo, sequenceNumber);
+        }
+
+        int sequenceNumber = accountNumbers.get(adhaarNo);
```
Comment (harry8906) [inline src/main/java/com/bluesky/bankapp/BankApplication.java:60]: You created sequenceNumber variable twice. You could have reused it.

```
int seqNumber = 1; // default value
if(!!accountNumbers.containsKey(adhaarNo)) {
     map.put....
} else {
     seqNumber = map.get(...) + 1;
     map.put(userId, seqNumber);
}
// use seqNum here...
```

File: src/main/java/com/bluesky/bankapp/BankApplication.java [hunk 3]
```diff
@@ -10,58 +12,179 @@
...
+        return map.get(accountNo);
+    }
+
     public static void main(String[] args) {
 
-        int choice;
-        Scanner scan = new Scanner(System.in);
+        int Choice;
+
+        String userName;
```
Comment (harry8906) [inline src/main/java/com/bluesky/bankapp/BankApplication.java:74]: In Java, unlike C, you should always declare variables when you  need to immediately use them.

You can move related variables to switch:case:1

File: src/main/java/com/bluesky/bankapp/BankApplication.java [hunk 4]
```diff
@@ -10,58 +12,179 @@
...
+                    adhaarNo = new String(scan.nextLine());
+
+                    System.out.println("Enter Balance:");
+                    balance = scan.nextInt();
+
+                    User user = new User(userName, DOB, mobileNo, adhaarNo, balance);
+                    account.addUserAccount(user);
+
+                    //Get Account number
+//                    System.out.println("get account number!");
```
Comment (harry8906) [inline src/main/java/com/bluesky/bankapp/BankApplication.java:148]: Never keep commented code in the repo, remove it.

File: src/main/java/com/bluesky/bankapp/BankApplication.java [hunk 5]
```diff
@@ -10,58 +12,179 @@
...
                     break;
                 }
 
                 case 2: {
 
-                    System.out.println("Removing Account for the particular user!");
+                    System.out.println("Enter Account number to remove from list");
+                    scan.nextLine();
+                    accountNo = new String(scan.nextLine());
+                    if (account.map.containsKey(accountNo)) {
```
Comment (harry8906) [inline src/main/java/com/bluesky/bankapp/BankApplication.java:175]: Here you have used BankApplication object to store the application data. Ideally there should be another component (can a single class) that handles this responsibility.

You can create a class BankAppDataStore which can contain the two maps you have created and there can be public getters to get those maps.

Have a instance of this class in BankApplication to store data.

File: src/main/java/com/bluesky/bankapp/BankApplication.java [hunk 6]
```diff
@@ -10,58 +12,179 @@
...
+                    } else {
+                        System.out.println("Account is not present!");
+                    }
                     break;
-                }
 
+                }
                 case 3: {
-                    System.out.println("Money transfering from one account to other!");
+                    System.out.println("Enter account to which you wanted to send money:");
```
Comment (harry8906) [inline src/main/java/com/bluesky/bankapp/BankApplication.java:190]: This is not covered in this issue/task. Please delete this code. You should only implement/make changes which are specified by the issue/task/jira.

File: src/main/java/com/bluesky/bankapp/BankApplication.java [hunk 7]
```diff
@@ -10,58 +12,179 @@
...
+                        System.out.println("Insufficient Balance!");
+                    }
+                   account.map.put(senderAccount, sender);
+                    account.map.put(receiverAccount, receiver);
                     break;
                 }
                 case 4: {
-                   // System.out.println(map);
-                    System.out.println("Display all records");
+                    System.out.println("Enter Account number:");
```
Comment (harry8906) [inline src/main/java/com/bluesky/bankapp/BankApplication.java:223]: Changed out of scope of this task, please remove the changes.

File: src/main/java/com/bluesky/bankapp/BankApplication.java [hunk 8]
```diff
@@ -10,58 +12,179 @@
...
+                    }
+
                     break;
-                }
 
+                }
                 case 5: {
 
-                    System.out.println("Exit!");
+                    //    System.out.println(map);
```
Comment (harry8906) [inline src/main/java/com/bluesky/bankapp/BankApplication.java:249]: Again, out of scope, please delete. In case you must need this code for listing account to test your other code, you should write test cases (we will cover this going forward.). For now if you want, you can keep this code.

File: src/main/java/com/bluesky/bankapp/BankApplication.java [hunk 9]
```diff
@@ -10,58 +12,179 @@
...
+                    for (Map.Entry<String, User> entry : account.map.entrySet()) {
+                        User account1 = entry.getValue();
+                        System.out.println(account1.accountNo + "  \t" + account1.userName + "  \t  " + account1.DOB + "  \t " + account1.mobileNo + "  \t" + account1.adhaarNo + "  \t" + account1.balance + "  \t\n");
+                    }
                     break;
                 }
-                default:
-                    throw new IllegalStateException("Unexpected value: " + choice);
-            }
+                case 6:{
```
Comment (harry8906) [inline src/main/java/com/bluesky/bankapp/BankApplication.java:257]: You can write a nice msg for user like - Thanks for user our bank, have a nice rest of the day!!"

File: src/main/java/com/bluesky/bankapp/User.java [hunk 10]
```diff
@@ -0,0 +1,58 @@
+package com.bluesky.bankapp;
+
+
+public class User {
+
+    public String userName;
+    public String DOB;
```
Comment (harry8906) [inline src/main/java/com/bluesky/bankapp/User.java:8]: Not following basic Java naming conventions might break flows of some of the automated flows in build sysmtems (not covered yet). You should stick with - birthDate

File: cls [hunk 11]
```diff
@@ -0,0 +1,50 @@
+[33mcommit 10153a67674a844a0142acebb0f58e0a0a1e6c0f[m[33m ([m[1;36mHEAD -> [m[1;32mtask-2[m[33m, [m[1;32mmaster[m[33m)[m
```
Comment (harry8906) [inline cls:1]: not sure what this is! Seems like an unwanted file. Please remove this file and push the changes.

File: src/main/java/com/bluesky/bankapp/BankApplication.java (hunk 5 above)
Comment (Asmita04) [reply src/main/java/com/bluesky/bankapp/BankApplication.java:175]: Yes , I thought of same .I will do this in task-3 .

Comment (harry8906) [reply src/main/java/com/bluesky/bankapp/BankApplication.java:175]: Should be part of this PR. Another task will be focused on  solving different problem.

Should be fixed in this PR.

File: src/test/java/com/bluesky/bankapp/BankApplicationTest.java [hunk 12]
```diff
@@ -1,7 +0,0 @@
-package com.bluesky.bankapp;
-
```
Comment (harry8906) [inline src/test/java/com/bluesky/bankapp/BankApplicationTest.java:2]: Why was this removed? you can keep this as it was.
